ipc/observability/src/traces.rs

```diff
@@ -22,32 +25,39 @@ pub fn create_temporary_subscriber() -> Subscriber {
 }
 
 // Sets a global tracing subscriber with the given configuration. Returns a guard that can be used to drop the subscriber.
-pub fn set_global_tracing_subscriber(config: &TracingSettings) -> Option<WorkerGuard> {
-    let console_filter = match &config.console {
-        Some(console_settings) => console_settings.level_to_filter(),
-        None => level_to_filter(&None),
-    };
-
-    // log all traces to stderr (reserving stdout for any actual output such as from the CLI commands)
-    let console_layer = fmt::layer()
-        .with_writer(std::io::stderr)
-        .with_target(false)
-        .with_file(true)
-        .with_line_number(true)
-        .with_filter(console_filter);
-
-    let (file_layer, file_guard) = match &config.file {
+pub fn set_global_tracing_subscriber(
+    config: &TracingSettings,
+) -> Option<(WorkerGuard, WorkerGuard)> {
+    let (logs_file_layer, traces_file_layer, file_guards) = match &config.file {
```

Comment:
The complexity of this method is too large. It's really hard to follow what it does, and we could document it better as well. I'm going to take a stab at improving this.